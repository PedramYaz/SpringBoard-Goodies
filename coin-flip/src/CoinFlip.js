import React, { useState } from "react";
import "./CoinFlip.css";
import imgHeads from "./imgHeads.jpeg";
import imgTails from "./imgTails.jpeg";

function CoinFlip() {
  const flipIdx = () => Math.floor(Math.random() * 2) + 1;

  const [totalFlips, setTotalFlips] = useState(0);
  const [headFlips, setHeadFlips] = useState(0);
  const [tailFlips, setTailFlips] = useState(0);
  const [flip, setFlip] = useState(flipIdx());

  return (
    <div className="CoinFlip">
      <h1 className="CoinFlip-title">Let's flip a coin!</h1>
      <img
        src={flip === 1 ? imgHeads : imgTails}
        className="CoinFlip-img"
        alt="coin"
      />
      <div className="CoinFlip-button-div" onClick={() => setFlip(flipIdx())}>
        <button
          onClick={
            flip === 1
              ? () => setTotalFlips(totalFlips + 1)
              : () => setTotalFlips(totalFlips + 1)
          }
          onClickCapture={
            flip === 1
              ? () => setHeadFlips(headFlips + 1)
              : () => setTailFlips(tailFlips + 1)
          }
          className="CoinFlip-button"
          data-testiqd="CoinFlip-button"
        >
          FLIP ME!
        </button>
      </div>
      <p data-testid="CoinFlip-text">
        Out of {totalFlips} flips, there have been {headFlips} heads and{" "}
        {tailFlips} tails
      </p>
    </div>
  );
}

export default CoinFlip;
