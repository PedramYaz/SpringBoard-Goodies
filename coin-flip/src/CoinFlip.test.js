import React from "react";
import { render, fireEvent, screen, getByText } from "@testing-library/react";
import CoinFlip from "./CoinFlip";

it("renders the function", function () {
  render(<CoinFlip />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<CoinFlip />);
  expect(asFragment()).toMatchSnapshot();
});

it("works when you click on flip me", function () {
  render(<CoinFlip />);

  const textInfo = screen.getByText("Out of", { exact: false });
  const noTextInfo = screen.getByText("Out of 0 flips", { exact: false });

  fireEvent.click(screen.getByText("FLIP ME!"));

  expect(textInfo).toBeInTheDocument();
  expect(noTextInfo).not.toBeInTheDocument();
});

// it("works when you click on flip me", function () {
//   const { getByText } = render(<CoinFlip />);

//   const textInfo = getByText("Out of");
//   const noTextInfo = getByText("0");

//   fireEvent.click(getByText("FLIP ME!"));

//   expect(textInfo).toBeInTheDocument;
//   expect(noTextInfo).not.toBeInTheDocument;
// });
