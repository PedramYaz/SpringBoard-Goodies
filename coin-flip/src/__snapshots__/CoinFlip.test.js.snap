// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CoinFlip"
  >
    <h1
      class="CoinFlip-title"
    >
      Let's flip a coin!
    </h1>
    <img
      alt="coin"
      class="CoinFlip-img"
      src="imgTails.jpeg"
    />
    <div
      class="CoinFlip-button-div"
    >
      <button
        class="CoinFlip-button"
        data-testid="CoinFlip-button"
      >
        FLIP ME!
      </button>
    </div>
    <p
      data-testid="CoinFlip-text"
    >
      Out of 0 flips, there have been 0 heads and 0 tails
    </p>
  </div>
</DocumentFragment>
`;
